kuzzleio/kuzzle
# Fix refresh propagation for document:delete

## What does this PR do ?

Propagate the `refresh` option to the storage engine.

### How should this be manually tested?

Copy this code
```
await sdk.document.create('index', 'collection', {}, null, { refresh: 'wait_for' })
await sdk.document.create('index', 'collection', {}, 'id', { refresh: 'wait_for' })

await sdk.document.delete('index', 'collection', 'id', { refresh: 'wait_for' })
let res = await sdk.document.search('index', 'collection', {})
console.log(res.hits.map(({_id}) => _id))
```

Then run Kourou with `kourou sdk:execute --editor` and paste it: you should see only one ID

### Boyscout

 - asyncify
